src/engine/sparqlExpressions/GroupConcatExpression.h

```diff
@@ -58,6 +58,9 @@ class GroupConcatExpression : public SparqlExpression {
     return std::visit(impl, std::move(childRes));
   }
 
+  // Required when using the hash map optimization.
+  std::string getSeparator() const { return separator_; }
```

Comment:
```suggestion
  const std::string& getSeparator() const { return separator_; }
```
The typical way for a const getter, the caller should decide if the copy is needed.